### ISSUE TITLE
[Upgrade] Upgrade never completes as it is missing the komodo image stream

### ISSUE THREAD

**Issue by avano** [syndesisio/fuse-online-install#170]
*******************
EDIT:
`1.7.23` with cherry-picked ac17a6087197425ab572d7698c74cb6f7b49ddd4 , see https://github.com/syndesisio/fuse-online-install/issues/170#issuecomment-518529554
*******************

`1.7.22`:

```
./update_ocp.sh                                                                                                                      
./update_ocp.sh: line 194: $2: unbound variable
Update imagestreams in avano-upgrade
Importing <registry>/fuse7/fuse-ignite-server:1.4-14 to fuse-ignite-server:1.7
Importing <registry>/fuse7/fuse-ignite-server:1.4-14 to fuse-ignite-server:1.7.22
Importing <registry>/fuse7/fuse-ignite-ui:1.4-6 to fuse-ignite-ui:1.7
Importing <registry>/fuse7/fuse-ignite-ui:1.4-6 to fuse-ignite-ui:1.7.22
Importing <registry>/fuse7/fuse-ignite-meta:1.4-13 to fuse-ignite-meta:1.7
Importing <registry>/fuse7/fuse-ignite-meta:1.4-13 to fuse-ignite-meta:1.7.22
Importing <registry>/fuse7/fuse-ignite-s2i:1.4-13 to fuse-ignite-s2i:1.7
Importing <registry>/fuse7/fuse-ignite-s2i:1.4-13 to fuse-ignite-s2i:1.7.22
Importing <registry>/fuse7/fuse-ignite-postgres_exporter:1.4-3 to fuse-ignite-postgres_exporter:1.7
Can't import

```

As you can see in the exporter:
`<registry>/fuse7/fuse-ignite-postgres_exporter:1.4-3`
- the correct namespace should be `fuse7-tech-preview`
- the correct image name is `fuse-postgres-exporter`

`to fuse-ignite-postgres_exporter:1.7`
- the correct image stream name is `postgres_exporter`

**Comment by avano** [comment]
@heiko-braun fyi

**Comment by heiko-braun** [comment]
@avano thanks

**Comment by lgarciaaco** [comment]
We stopped doing few things the `update_ocp.sh` script relies on:
 - We shouldn't call docker images with `-` as part of the name, ideally they are called with a simple word, like `server`, `meta` and so on. When we start using wordings like `komodo-server` or `postgres-exporter` it all gets nasty. We need a simpler name for these images. I would suggest to call komodo just `komodo`, and postgres exporter `pexporter`.
 - The script assumes that docker images are called $PREFIX + $NAME, where prefix is `fuse-ignite` and name is `meta`, `server` AND `komodo`, the docker image should be called: `fuse-ignite-meta`, `fuse-ignite-server` AND `fuse-ignite-komodo` (which is called _data-virtualization-server-rhel7:$tag_, and for postgres exporter _fuse-postgres-exporter:$tag_). We need the same image prefix for komodo and postgres exporter than the rest of the images.

I could work around these issues without changing the image names. We have to go back to the mentioned standards.

**Comment by heiko-braun** [comment]
> We need the same image prefix for komodo and postgres exporter than the rest of the images.

I am afraid this is not possible at this stage, and I don't think we should make assumption about the image name prefixes in the script. Why no simply call them out fully?

**Comment by lgarciaaco** [comment]
> > We need the same image prefix for komodo and postgres exporter than the rest of the images.
> 
> I am afraid this is not possible at this stage, and I don't think we should make assumption about the image name prefixes in the script. Why no simply call them out fully?

Sure it wont be possible at this stage, but why not give things simple names like the ones I recommended? That shouldn't be an issue for next release or so.

There is a relation between name and few other elements: for name `server`, the image tag is `tag_server` [here](https://github.com/syndesisio/fuse-online-install/blob/master/fuse_online_config.sh#L8) ), and the the name of the ImageStream (`fuse-ignite-server`) and the image itself  (`registry/repo/fuse-ignite-server`) are build based on that prefix and the name [here](https://github.com/syndesisio/fuse-online-install/blob/1.7/update_ocp.sh#L331)

**Comment by heiko-braun** [comment]
No objection for simple names

**Comment by lgarciaaco** [comment]
@avano can you please test ^^

**Comment by lgarciaaco** [comment]
After thinking and talking about this, we don't include external bits in the upgrade process. Removing komodo and the exporter should fix the problem

**Comment by avano** [comment]
@lgarciaaco @heiko-braun removing the datavirt image stream now causes the upgrade pod to wait indefinitely, as it is waiting for komodo-server to be scaled to 1:

```
=== * Restart all deployments (upgrade_60_restart_all)
deploymentconfig.apps.openshift.io/komodo-server scaled
deploymentconfig.apps.openshift.io/syndesis-db scaled
deploymentconfig.apps.openshift.io/syndesis-meta scaled
deploymentconfig.apps.openshift.io/syndesis-oauthproxy scaled
deploymentconfig.apps.openshift.io/syndesis-prometheus scaled
deploymentconfig.apps.openshift.io/syndesis-server scaled
deploymentconfig.apps.openshift.io/syndesis-ui scaled
Waiting for komodo-server to be scaled to 1
```

and obviously since there is no image stream for komodo, it waits forever and the upgrade never completes.

also leaving out the datavirt server from the upgrade means that user can't use datavirt when upgrading from 7.3.1 to 7.4

**Comment by heiko-braun** [comment]
@lgarciaaco I guess this means we need to include komodo at least

**Comment by lgarciaaco** [comment]
https://github.com/syndesisio/fuse-online-install/pull/180 removes some code that was needed for previous upgrade, but not needed anymore. It was creating the IS for komodo. I think the upgrade pod wont be trying for it anymore.

**Comment by lgarciaaco** [comment]
@avano can you cherry pick and try?

**Comment by avano** [comment]
fixed on 1.7.x branch

**Comment by heiko-braun** [comment]
Good work, thanks guys!
